domain/storageprovisioning/state/base_test.go

```diff
@@ -23,6 +25,8 @@ import (
 // tests. Base suite does not seed a starting state and does not run any tests.
 type baseSuite struct {
 	schematesting.ModelSuite
+
+	storageInstCount int
```

Comment:
Can you just use the sequence table like we do in the state layer for this? There is a domain and state for getting the next int. Having this will just lead to flaky testing.